design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,260 @@
+# Kubevirt Feature Lifecycle
+
+## Summary
+
+Kubevirt requires a process with a clear policy on how features
+are introduced, evaluated and finally graduated or dropped.
+
+This proposal defines the steps and policies to follow in order
+to manage a feature (lifecycle) in Kubevirt.
+
+## Overview
+
+### Motivation
+Kubevirt has grown into a matured virtualization management solution
+with a large set of features.
+
+New features are being proposed and added regularly to its portfolio.
+
+With time, the challenge of supporting and maintaining such a large
+set of features raised the need to re-examine their relevance.
+It also raised the need to examine with more care features graduation.
+
+The Kubevirt community has tried to control the flow of features
+informally through feature-gates, similar to Kubernetes.
+However, as time passed, this has been found to be insufficient due
+to the lack of a well-defined processes and policy on how a feature
+should progress in its lifecycle.
+
+Eventually we would like to see features being evaluated carefully
+before they are introduced, while they are experimented with and
+proven to be actually in use (and useful) before graduating.
+
```

Comment:
In this section, please include a reference to semver.

semver is constraining specifically the deprecation.